avalanche/benchmarks/scenarios/rl_scenario.py

```diff
@@ -9,14 +9,108 @@
 # Website: avalanche.continualai.org                                           #
 ################################################################################
 """Reinforcement Learning scenario definitions."""
-from typing import Union, Sequence
+from avalanche.benchmarks.scenarios import CLExperience, ExperienceAttribute, CLScenario, EagerCLStream
+from typing import Callable, List, Union, Dict
+import numpy as np
+import torch
+import random
 
 try:
-    from gym import Env
+    from gym import Env, Wrapper
 except ImportError:
-    # empty class to make sure everything below works without changes
+    # empty classes to make sure everything below works without changes
     class Env:
         pass
+    class Wrapper:
+        pass
+
+
+class RLExperience(CLExperience):
+
+    def __init__(self, env: Union[Env, Callable[[], Env]], n_envs: int = 1, task_label: int = None, current_experience: int = None, origin_stream=None):
```

Comment:
Callable[[], Env] is needed to create environments lazily? Isn't it better to simplify this definition and make the entire stream lazy?

Reply:
We also need to support environments which handle the change of task internally (e.g. habitat) and whose instance must persist across experiences; I'd rather leave it as is for now

Reply:
makes sense, let's keep it as is.